irods/test/data_obj_test.py

```diff
@@ -654,6 +656,61 @@ def test_multiple_reads(self):
             with obj.open('r') as f:
                 self.assertEqual(f.read().decode(), obj.path)
 
+    def _skip_unless_connected_to_this_computer_by_other_than_localhost_synonym(self):
+        if self.sess.host not in ({socket.gethostname()} - {'localhost', '127.0.0.1'}):
+            self.skipTest('This test requires being connected to a local server, but not via "localhost" or a synonym.')
+
+    def test_redirect_in_data_object_put__issue_452(self):
+        self._skip_unless_connected_to_this_computer_by_other_than_localhost_synonym()
+        paths_to_delete = []
+        with self.create_simple_resc(hostname = 'localhost') as rescName:
+            try:
+                SMALL = 1; LARGE = 40
+                for megabyte_length in (SMALL, LARGE):
+                    with NamedTemporaryFile() as f:
+                        megabyte = b'_'*1024**2
+                        for _ in range(megabyte_length):
+                            f.write(megabyte)
+                        f.flush()
```

Comment:
Guess that would work....will try.  That results in an all-nulls file, yes?

Reply:
well, more like... 'we don't know what's in there' as it just marks the existing memory as 'part of this file'.    so... make sure you're testing what you think you're testing and you should be okay.   but it's definitely a lot faster...

Reply:
I have created several scripts to test various sized files for stress testing various iRODS setups, and I found the most efficient methods for creating files with and without specific information in them.  For a file with only null values (yes, actual null values if you look at it with hexedit), `os.truncate()` is the quickest, but the file _must_ exist:

```python
with open(filename, "wb") as fd:
    pass
os.truncate(filename, size)
```

But if you need specific header information, for instance, a different method is faster:

```python
with open(filename, "wb") as fd:
    fd.write(header)
    fd.seek(size - 1)
    fd.write(b"\0")
```

If you were to expand the file with `truncate()` instead of the last `seek()`/`write()` here, you end up writing the file twice with the associated overhead.

Reply:
Superceded, as we are now examining the contents for purposes of the test.